File: Practice/17/Python/seventeenthTask.py
def get_numbers():
    return 37


def get_k():
    return 12


count = {}
last_round = {}
for i in range(get_numbers()):
    count[i] = last_round[i] = 0
highest_number = round_now = reds = blacks = 0
while (True):
    round_now += 1
    inp = int(input())
    if inp < 0:
        break
    count[inp] += 1
    last_round[input] = round_now
    highest_number = max(count.values())
    for i in range(get_numbers()):
        if count[i] == highest_number:
            print(i, end=' ')
    print()
    for i in range(get_numbers()):
        if last_round[i] == 0 or round_now - last_round[i] > get_k():
            print(i, end=' ')
    print()
    if inp != 0:
        if inp <= 10 or (inp > 18 and inp <= 28):
            if inp % 2 == 1:
                reds += 1
            else:
                blacks += 1
        else:
            if inp % 2 == 0:
                reds += 1
            else:
                blacks += 1

    print(reds, blacks, '\n')

File: Practice/17/C++/seventeenthTask/seventeenthTask/seventeenthTask.cpp
#include <iostream>
using namespace std;
const int numbers = 37;
const int k = 12;

int main() {
    int count[numbers];
    int last_round[numbers];
    for (int i = 0; i < numbers; i++)
        count[i] = last_round[i] = 0;
    
    int input;
    int round_now = 0;
    int blacks = 0, reds = 0;
    while (++round_now) {
        cin >> input;
    	if (input < 0)
    	    break;
    	    
        count[input]++;
        last_round[input] = round_now;
            
        int highest_number = -1;
    	for (int i = 0; i < numbers; i++)
    	    if (count[i] > highest_number)
    	        highest_number = count[i];
    	
    	for (int i = 0; i < numbers; i++)
            if (count[i] == highest_number)
                cout << i << " ";
    	cout << endl;
    
    	for (int i = 0; i < numbers; i++)
    	    if (last_round[i] == 0 or (round_now - last_round[i] > k))
                cout << i << " ";
    	cout << endl;
            
    	if (input != 0) {
            if (input <= 10 or (input > 18 and input <= 28))
                if (input % 2 == 1) ++reds;
                else ++blacks;
            else
                if (input % 2 == 0) ++reds;
                else ++blacks;
    	}

        cout << reds << " " << blacks << endl;
        cout << endl;
    }
}